### ISSUE TITLE
Frame isn't being displayed back when leaving a delve

### ISSUE THREAD

**Issue by sampconrad** [BelegCufea/CompassHUD#11]
The frame properly hides itself when zoning into a Delve, but sometimes when zoning back out, it's not being displayed. It requires a reload to fix it.

**Comment by BelegCufea** [comment]
It is strange as it should stay visible in Delves.
I run only few Delves myself, but it was always visible. I run it solo though.

Does it happen when only CompassHUD is installed?

**Comment by sampconrad** [comment]
Did some testing and seems to be very hit-or-miss. Some Delves it manages to hide the compass properly, and some don't.

**Comment by sampconrad** [comment]
I think you're running into the same issue I did with my Combat Mode addon.

You know those quick loading screens when starting a m+ run, where it quickly shows a splash screen while they teleport everyone to a spot and start the m+ timer? Well, those loading screens don't fire the ``PLAYER_ENTERING_WORLD`` event like every other loading screen does, instead they fire the ``LOADING_SCREEN_ENABLED`` event and once done fire the ``LOADING_SCREEN_DISABLED``.

I suspect Delves are doing something similar, though I have no way of testing it right now.

Maybe start with that? Check which event is firing and use that for the callback to toggle the compass.

EDIT:

See how on your OnEvent loop, the instance check only happens when the ``PLAYER_ENTERING_WORLD`` event fires? If Delves don't fire that event, as I suspect they don't, then it would explain why it's bugging out. If I'm correct in that they fire the ``LOADING_SCREEN_DISABLED`` event instead, then it should It should also be checking for that:
```
    if event == "PLAYER_ENTERING_WORLD" then
        local _, instanceType = IsInInstance()
        player.instance = instanceType
        if player.instance == "none" then
            HUD:Show()
            HUD:SetScript('OnUpdate', onUpdate)
        else
            HUD:Hide()
            HUD:SetScript('OnUpdate',nil)
        end
    end
```

**Comment by sampconrad** [comment]
So I tested it and it's firing these events when leaving a Delve:
```ZONE_CHANGED_NEW_AREA, AREA_POIS_UPDATED, FOG_OF_WAR_UPDATED, PLAYER_MAP_CHANGED```

**Comment by BelegCufea** [comment]
Well, compass works in Delves. I can hide it through ZONE_CHANGED_NEW_AREA event. But should I?

Edit: Well, my bad. It does not work there.

**Comment by BelegCufea** [comment]
Will add an option :-)

**Comment by BelegCufea** [comment]
Should be fixed in v1.2.3
